src/Utilities/FlowAnalysis/Framework/DataFlow/AbstractBlockAnalysisResult.cs

```diff
@@ -1,17 +1,39 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+
 namespace Microsoft.CodeAnalysis.FlowAnalysis.DataFlow
 {
     /// <summary>
     /// Result from execution of a <see cref="DataFlowAnalysis"/> on a basic block.
     /// </summary>
-    internal abstract class AbstractBlockAnalysisResult
+    internal abstract class AbstractBlockAnalysisResult : IDisposable
     {
         protected AbstractBlockAnalysisResult(BasicBlock basicBlock)
         {
             BasicBlock = basicBlock;
         }
 
         public BasicBlock BasicBlock { get; }
+
+        public bool IsDisposed { get; private set; }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            IsDisposed = true;
+        }
+
+#pragma warning disable CA1063 // Implement IDisposable Correctly - We want to ensure that we cleanup managed resources even when object was not explicitly disposed.
+        ~AbstractBlockAnalysisResult()
```

Comment:
:exclamation: let's have a discussion about this in our next design meeting

Reply:
Sure, there is no rush to get this in and I also want to re-evaluate alternate ways to tackle the performance challenges of analyzing huge method bodies - for example, [CSharpCommandLineParser.Parse](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/CommandLine/CSharpCommandLineParser.cs,293eb00017c0ffd4) generates a CFG with 500+ basic blocks and ~200 locals/parameters and generating/cloning dictionary(s) per basic block with 200+ key/value pairs and then merging data values across basic blocks is very expensive, especially when done iteratively till it reaches a fixed point. The current analysis performance is fine for reasonable sized methods.